sim/pokemon.ts

```diff
@@ -50,10 +50,10 @@ export class Pokemon {
 
 	baseTemplate: Template;
 	template: Template;
-	speciesData: AnyObject;
+	speciesData: AnyObjectWithID;
```

Comment:
I was thinking about removing the ID, actually.

Reply:
(I don't think we've ever needed to read it in the past eight years of supporting it.)

Reply:
Removed `AnyObjectWithID` typing. Actually removing the `id: ''` field is better saved for a future change (maybe one which also renames them to `fooState` like I believe you requested in the past?)

Reply:
I approve of that.